geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/api/GeoMesaIndexManager.scala

```diff
@@ -50,12 +50,27 @@ trait GeoMesaIndexManager[O <: GeoMesaDataStore[O, F, W], F <: WrappedFeature, W
   def setIndices(sft: SimpleFeatureType): Unit = {
     import org.locationtech.geomesa.utils.geotools.RichSimpleFeatureType.RichSimpleFeatureType
     // remove the enabled indices after fetch so we don't persist them
-    val enabled = Option(sft.getUserData.remove(SimpleFeatureTypes.Configs.ENABLED_INDICES))
-    val indices: Seq[GeoMesaFeatureIndex[O, F, W]] = enabled match {
-      case None => CurrentIndices
-      case Some(e) => e.toString.split(",").map(_.trim).flatMap(n => CurrentIndices.find(_.name.equalsIgnoreCase(n)))
+    val enabled = Option(sft.getUserData.remove(SimpleFeatureTypes.Configs.ENABLED_INDICES)).filter(_.toString.length > 0)
+    val indices: Seq[(String, Int, IndexMode)] = enabled match {
+      case None => CurrentIndices.collect { case i if i.supports(sft) => (i.name, i.version, IndexMode.ReadWrite) }
+      case Some(e) =>
+        e.toString.split(",").map(_.trim).map { name =>
+          val index = CurrentIndices.find(_.name.equalsIgnoreCase(name)).getOrElse {
```

Comment:
do you think there'd be any issue upgrading an old 1.2.x datastore or codeset to the new ones...i will try to give that a run. in case something like ```attr``` is now ```attr_v2``` or something.